### PR TITLE
Built-in browser input validation & NumberInput validation

### PR DESCRIPTION
**Pull request by peolivei2**
`onchange` events are not triggered for bad inputs, which caused our TextInput component to lose track of the input value whenever the value entered by the user did not match the input type. One of the side effects is that a user can add an arbitrary number of dashes and periods to number fields.

In the NumberInput side, our validation for `onkeydown` relied on the component state having the most recent value, which is not true until `onchange` triggers for a valid input value, thus the checks to avoid multiple periods and dashes did not work in most cases.

### CODE REVIEW

**Comment by misteinb** [comment]
Which browsers did you test this on?

**Comment by peolivei2** [comment]
@misteinb Chrome, FF, Edge

**Comment by vincenzocaruso** [comment]
We should remove all the custom validation done on this controls and let the consumer to add the validation on his side.

File: lib/components/Input/NumberInput.tsx [hunk 1]
```diff
@@ -111,15 +111,13 @@ export class NumberInput extends React.Component<NumberInputProps, NumberInputSt
             !this.isPositive()
             /** Firefox uses a different keycode for dashes (-) than other browsers... */
             && (event.keyCode === keyCode.dash || event.keyCode === keyCode.firefoxDash)
```
Comment (dluc) [inline lib/components/Input/NumberInput.tsx]: please add a comment to docs about FF, e.g. http://www.javascripter.net/faq/keycodes.htm

**Comment by vincenzocaruso** [comment]
we need to remove all the custom validations that we have right now

**Pull request closed without merging**